cmd/config/apply_test.go

```diff
@@ -0,0 +1,333 @@
+	}
+	err = registerResponder(deleteExpected, http.StatusNoContent, "/roles/test2", http.MethodDelete)
+	if err != nil {
+		suite.T().Fatalf("TestDeployStartJsonSuccess failed with: %s", err)
+	}
+
+	tempFile := util.TempAppFile("", "app", testConfigYamlStrForDeleteAllowAutoDelete)
+	if tempFile == nil {
+		suite.T().Fatal("TestDeployStartJsonSuccess failed with: Could not create temp app file.")
+	}
+	suite.T().Cleanup(func() { os.Remove(tempFile.Name()) })
+	outWriter := bytes.NewBufferString("")
+	cmd := getConfigApplyCmdWithTmpFile(outWriter, tempFile, "json")
+	err = cmd.Execute()
+	if err != nil {
+		suite.T().Fatalf("TestDeployStartJsonSuccess failed with: %s", err)
+	}
+	_, err = ioutil.ReadAll(outWriter)
+	if err != nil {
+		suite.T().Fatalf("TestDeployStartJsonSuccess failed with: %s", err)
+	}
+	callCount := httpmock.GetCallCountInfo()
+	suite.Equal(1, callCount["DELETE /roles/test2"])
+	suite.Equal(1, callCount["GET /roles"])
+	suite.Equal(1, callCount["PUT /roles/test"])
+}
+
+func (suite *ConfigApplyTestSuite) TestConfigApplyDeleteRoleDontAllowAutoDelete() {
+	getExpected := []model.RoleConfig{
+		{
+			Name:   "test",
+			Tenant: "testTenant",
+			Grants: []model.GrantConfig{
+				{
+					Type:       "api",
+					Resource:   "org",
+					Permission: "all",
+				},
+			},
+		},
+		{
+			Name:   "test2",
+			Tenant: "testTenant",
+			Grants: []model.GrantConfig{
+				{
+					Type:       "api",
+					Resource:   "org",
+					Permission: "all",
+				},
+			},
+		},
+	}
+	putExpected := model.RoleConfig{
+		Name:   "test",
+		Tenant: "testTenantNew",
+		Grants: []model.GrantConfig{{
+			Type:       "api",
+			Resource:   "org",
+			Permission: "all",
+		},
+		}}
+	err := registerResponder(getExpected, http.StatusOK, "/roles", http.MethodGet)
+	if err != nil {
+		suite.T().Fatalf("TestDeployStartJsonSuccess failed with: %s", err)
+	}
+	err = registerResponder(putExpected, http.StatusOK, "/roles/test", http.MethodPut)
+	if err != nil {
+		suite.T().Fatalf("TestDeployStartJsonSuccess failed with: %s", err)
+	}
+
+	tempFile := util.TempAppFile("", "app", testConfigYamlStrForDeleteDontAllowAutoDelete)
+	if tempFile == nil {
+		suite.T().Fatal("TestDeployStartJsonSuccess failed with: Could not create temp app file.")
+	}
+	suite.T().Cleanup(func() { os.Remove(tempFile.Name()) })
+	outWriter := bytes.NewBufferString("")
+	cmd := getConfigApplyCmdWithTmpFile(outWriter, tempFile, "json")
+	err = cmd.Execute()
+	if err != nil {
+		suite.T().Fatalf("TestDeployStartJsonSuccess failed with: %s", err)
+	}
+	_, err = ioutil.ReadAll(outWriter)
+	if err != nil {
+		suite.T().Fatalf("TestDeployStartJsonSuccess failed with: %s", err)
+	}
+	callCount := httpmock.GetCallCountInfo()
+	suite.Equal(0, callCount["DELETE /roles/test2"])
+	suite.Equal(1, callCount["GET /roles"])
+	suite.Equal(1, callCount["PUT /roles/test"])
+}
+
+func registerResponder(body interface{}, status int, url, method string) error {
```

Comment:
```suggestion
func registerResponder(body any, status int, url, method string) error {
```

@danielpeach tells me that `any` should be preferred over `interface{}`